net/seafile-server/Makefile

```diff
@@ -1,141 +1,146 @@
-PKG_RELEASE=$(PKG_SOURCE_VERSION)-5
+PKG_VERSION:=5.1.1
+PKG_RELEASE=$(PKG_SOURCE_VERSION)-6
 PKG_LICENSE:=GPL-3.0
 
 PKG_SOURCE_PROTO:=git
 PKG_SOURCE_URL:=https://github.com/haiwen/seafile.git
 PKG_SOURCE_SUBDIR:=$(PKG_NAME)-$(PKG_VERSION)
-PKG_SOURCE_VERSION:=c0166019c712e2e1d5e532fd5f7401b1b72db6d8
+PKG_SOURCE_VERSION:=5a8677a3acdde748caa387adfc054da64f2a24ea
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION)-$(PKG_SOURCE_VERSION).tar.gz
 PKG_FIXUP:=autoreconf
 PKG_INSTALL:=1
 
 include $(INCLUDE_DIR)/package.mk
 include $(INCLUDE_DIR)/nls.mk
+$(call include_mk, python-package.mk)
 
 define Package/seafile-server
     SECTION:=net
     CATEGORY:=Network
     TITLE:=Seafile server
     MAINTAINER:=Gergely Kiss <mail.gery@gmail.com>
     URL:=http://seafile.com/
-    DEPENDS:=+libarchive +libopenssl +glib2 +libsearpc +seafile-ccnet +seafile-seahub \
-		+sqlite3-cli +python-mysql +jansson +libevent2 +libevent2-openssl +zlib +libzdb +libsqlite3 \
-		+libmysqlclient +libpthread +libuuid \
-		+bash +sudo +procps-ng +procps-ng-pkill $(ICONV_DEPENDS)
-    EXTRA_DEPENDS:=seafile-ccnet (=4.1.2-a73109f09af4ecc49cdc4c57cdde51b38e15c31a-2), seafile-seahub (=4.1.2-3fb1288f920de03a4e2e6a06b60671ce98971742)
+    DEPENDS:=+libarchive +libopenssl +glib2 +libsearpc +seafile-ccnet +seafile-seahub +sqlite3-cli +python-mysql \
+		+jansson +libevent2 +libevent2-openssl +zlib +libzdb +libsqlite3 +libmysqlclient \
+		+libpthread +libuuid +bash +procps-ng +procps-ng-pkill +SEAFILE_FUSE_SUPPORT:libfuse $(ICONV_DEPENDS)
+    EXTRA_DEPENDS:=seafile-ccnet (=5.1.1-6d94fdc4296d542a5552039dc13a67c3448e8b13-1), seafile-seahub (=5.1.1-95f762f396c81f840fe658ab983df53784083bb1-1)
+    MENU:=1
 endef
 
-define Package/seafile-server/description
-   Open source cloud storage with advanced features on privacy protection and teamwork.
+define Package/seafile-server/config
+	source "$(SOURCE)/Config.in"
 endef
 
-define Package/seafile-server/conffiles
-/etc/config/seafile
+define Package/seafile-server/description
+   Open source cloud storage with advanced features on privacy protection and teamwork.
 endef
 
-CONFIGURE_ARGS += --disable-riak \
-		    --disable-client \
-		    --disable-fuse \
+CONFIGURE_ARGS += --disable-client \
 		    --enable-server \
 		    --enable-python \
 		    --disable-static-build \
-		    --disable-server-pkg \
-		    --disable-console
+		    --disable-server-pkg
 
-PKG_BUILD_DEPENDS:=vala/host \
-		    libsearpc/host \
-		    libevhtp-1.1
+ifeq ($(CONFIG_SEAFILE_FUSE_SUPPORT),y)
+	CONFIGURE_ARGS += --enable-fuse
+	TARGET_CFLAGS += -I$(STAGING_DIR)/usr/include/fuse
+else
+	CONFIGURE_ARGS += --disable-fuse
+endif
+
+ifeq ($(CONFIG_SEAFILE_CONSOLE_SUPPORT),y)
+	CONFIGURE_ARGS += --enable-console
+else
+	CONFIGURE_ARGS += --disable-console
+endif
 
+ifeq ($(CONFIG_SEAFILE_RIAK_SUPPORT),y)
+	CONFIGURE_ARGS += --enable-riak
+else
+	CONFIGURE_ARGS += --disable-riak
+endif
+
+PKG_BUILD_DEPENDS:=vala/host libevhtp
+
+TARGET_CFLAGS += -minterlink-mips16
```

Comment:
Yes, it is, otherwise the compilation fails with the following error:

/home/nb/openwrt/openwrt_trunk/staging_dir/toolchain-mips_34kc_gcc-5.3.0_musl-1.1.14/lib/gcc/mips-openwrt-linux-musl/5.3.0/../../../../mips-openwrt-linux-musl/bin/ld: upload-file.o: .text+0x110: Unsupported jump between ISA modes; consider recompiling with interlinking enabled.

This should not cause any issues, as I could successfully build the packages for a RPI which is a non-MIPS arch (brcm2708).